tool.go

```diff
@@ -91,6 +91,10 @@ func main() {
 	cmdBuild.Flags().AddFlag(flagTags)
 	cmdBuild.Flags().AddFlag(flagLocalMap)
 	cmdBuild.Run = func(cmd *cobra.Command, args []string) {
+		if err := verifyAndUnsetGOARCH(); err != nil {
```

Comment:
I've spent some time looking through https://godoc.org/github.com/gopherjs/gopherjs/build, https://godoc.org/github.com/gopherjs/gopherjs/compiler, https://godoc.org/github.com/gopherjs/gopherjs and the source, thinking about a better place where this could be moved.

I didn't find a great place.

There are multiple different entry points, depending on whether you're building import paths or specifying .go files on the command line, etc. Finding/catching them all is tricky and hard to be confident in.

`github.com/gopherjs/gopherjs/build.Session.BuildPackage` is pretty central and gets called by most entry points, but it seems a little too late. In some cases, a bunch of work loading and importing packages is done before `BuildPackage` is called.

`github.com/gopherjs/gopherjs/build.NewSession` seems more fitting, it's also central, and happens much earlier. But it doesn't return an error right now. We could change it to do that.

But it still happens inside a for loop in tool.go because of watcher support.

Also, nothing compilation-related happens inside `gopherjs doc` command. But I don't want it to do anything with non-`js` architectures (even if it happens to work, supporting that is out of scope).

In the end, checking that GOARCH is not some crazy value makes most sense to me inside `tool.go`, first thing. I don't want anything to happen if GOARCH=mips64 or something else. GopherJS only supports compiling GOARCH=js architecture, and there are no short term plans to change that.

Unless you have some other suggestion that I've missed, I'd prefer we keep this check inside tool, that happens right away, rather than later.